Revisit Multi Site Guide

pruivo opened the pull request:
<!---
Please read https://github.com/keycloak/keycloak/blob/main/CONTRIBUTING.md and follow these guidelines when contributing to Keycloak
-->

Closes #32745

pruivo committed:
Revisit Multi Site Guide

Closes #32745

Signed-off-by: Pedro Ruivo <pruivo@redhat.com>

pruivo commented:
It is easier to review changes here. I'll create a GHI in the future.

ryanemerson commented on docs/guides/high-availability/concepts-multi-site.adoc:
```diff
@@ -121,7 +121,7 @@ A synchronously replicated database ensures that data written in one site is alw
 
 Why synchronous {jdgserver_name} replication?::
-A synchronously replicated {jdgserver_name} ensures that sessions created, updated and deleted in one site are always available on the other site after a site failure and no data is lost.
+A synchronously replicated {jdgserver_name} ensures that cached data cin one site are always available on the other site after a site failure and no data is lost.
```
Type, `/cin/in`

pruivo replied:
I'm just checking if you guys are paying attention :rofl:

pruivo commented:
I've assigned this to an existing GHI: #32745

pruivo commented on docs/guides/high-availability/concepts-memory-and-cpu-sizing.adoc:
```diff
@@ -45,7 +45,7 @@ Most CPU time goes into creating new TLS connections, as each client runs only a
 * For each 120 refresh token requests per second, 1 vCPU to the cluster (tested with up to 435 refresh token requests per second).
 
 * Leave 200% extra head-room for CPU usage to handle spikes in the load.
```
`Leave 200% extra head-room` -> does it seem excessive?

mhajas commented on docs/guides/high-availability/concepts-memory-and-cpu-sizing.adoc:
```diff
@@ -44,8 +44,8 @@ Most CPU time goes into creating new TLS connections, as each client runs only a
 
-* Leave 200% extra head-room for CPU usage to handle spikes in the load.
-This ensures a fast startup of the node, and sufficient capacity to handle failover tasks like, for example, re-balancing Infinispan caches, when one node fails.
+* Leave 125% extra head-room for CPU usage to handle spikes in the load.
```
I am wondering what was the reason for the 200% head-room with A/P setup. If it was necessary there, maybe it is still needed for A/A as well. We should probably wait for @ahus1 's opinion

pruivo replied:
I don't remember where we got the 200%.
With A/P, you should not need any head-room at all. 
With A/A it would make sense to double the CPU count in case one of the sites crashes, the other needs to be able to handle the load. In any case, 200% seems way too pessimistic.

mhajas replied:
I agree with you and 125% makes sense to me. Maybe the reason for the extra headroom is some unexpected load, like some peak period but I am not sure that is something we should document, it is probably something the users should consider themselves based on their use-case. Let's wait for @ahus1. From my perspective this change is ok.

mhajas reviewed (commented):
Thank you for the changes @pruivo. I added two comments.

pruivo commented on docs/guides/high-availability/concepts-multi-site.adoc:
```diff
@@ -83,7 +83,7 @@ Manual operations might be necessary depending on the database.
 Some requests might receive an error message until the loadbalancer detects the failure.
 | No data loss^3^
-| Less than one minute
+| Less than two minutes
```
It is based on Kamesh results. Not only do you have to wait until JGroups removes the site from the view, then you also have additional time for Prometheus to scrape the metrics and trigger the alert.

mhajas replied:
Ok makes sense. Thanks

ahus1 committed:
Review: Re-adding the JGroups threads to the concepts as this document might be shared outside the HA context.

Also changing the extra CPU to 150% as this guide might be used in a single-site setup

Signed-off-by: Alexander Schwartz <aschwart@redhat.com>

ahus1 reviewed (approved):
Thank for removing the JGroups dependency on the Keycloak deployment, I fully agree on that like you explained in today's meeting: due to the external Infinispan no JGroups is used inside of Keycloak, and therefore no restriction applies. 

As the general threading guide might be read out of the context of multi-site, I resurrected this content until we move it to another place. 

Again for the sizing guide, we now use the middle ground of 150%.
